core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -788,6 +788,20 @@ abstract class RDD[T: ClassTag](
   }
 
   /**
+   * Applies a function to each partition of this RDD, while tracking the index
+   * of the original partition.
+   */
+  def foreachPartitionWithIndex(
```

Comment:
Can't this be done with just mapPartitionsWithContext followed by a foreach? I feel we are adding all kinds of random APIs to RDD that is making it very hard to maintain. If it is completely up to me, I'd even remove a bunch of existing RDD APIs ...